spec.html

```diff
@@ -40393,6 +40499,90 @@ <h1>`Promise.allSettled` Reject Element Functions</h1>
         </emu-clause>
       </emu-clause>
 
+      <emu-clause id="sec-promise.any">
+        <h1>Promise.any ( _iterable_ )</h1>
+        <p>The `any` function returns a promise that is fulfilled by the first given promise to be fulfilled, or rejected with an `AggregateError` holding the rejection reasons if all of the given promises are rejected. It resolves all elements of the passed iterable to promises as it runs this algorithm.</p>
+        <emu-alg>
+          1. Let _C_ be the *this* value.
+          1. Let _promiseCapability_ be ? NewPromiseCapability(_C_).
+          1. Let _promiseResolve_ be GetPromiseResolve(_C_).
+          1. IfAbruptRejectPromise(_promiseResolve_, _promiseCapability_).
+          1. Let _iteratorRecord_ be GetIterator(_iterable_).
+          1. IfAbruptRejectPromise(_iteratorRecord_, _promiseCapability_).
+          1. Let _result_ be PerformPromiseAny(_iteratorRecord_, _C_, _promiseCapability_, _promiseResolve_).
+          1. If _result_ is an abrupt completion, then
+            1. If _iteratorRecord_.[[Done]] is *false*, set _result_ to IteratorClose(_iteratorRecord_, _result_).
+            1. IfAbruptRejectPromise(_result_, _promiseCapability_).
+          1. Return Completion(_result_).
+        </emu-alg>
+        <emu-note>
+          <p>The `any` function requires its *this* value to be a constructor function that supports the parameter conventions of the `Promise` constructor.</p>
+        </emu-note>
+
+        <emu-clause id="sec-performpromiseany" aoid="PerformPromiseAny">
+          <h1>Runtime Semantics: PerformPromiseAny ( _iteratorRecord_, _constructor_, _resultCapability_, _promiseResolve_ )</h1>
+          <p>When the PerformPromiseAny abstract operation is called with arguments _iteratorRecord_, _constructor_, _resultCapability_ (a PromiseCapability Record), and _promiseResolve_, the following steps are taken:</p>
```

Comment:
Please format this into the standard "The abstract operation AO takes arguments _foo_ (a Foo) and _bar_ (a Bar). It performs the following steps when called:"

Reply:
Q: does this mean the textual explanation ("The `any` function returns a promise that is..") has to be removed completely? That'd be unfortunate, since having these non-normative text snippets describing the intention of an AO would make the spec much easier to understand.

Is it allowed to have:
The abstract operation AO takes arguments foo (a Foo) and bar (a Bar). It returns a promise that is etc etc. It performs the following steps when called:

Reply:
> does this mean the textual explanation ("The any function returns a promise that is..") has to be removed completely?

No, for two reasons: 1) so far this convention is only used for abstract operations, not built-in functions like `Promise.any` and 2) for abstract operations you can have a description in the middle just as you suggest - compare e.g. [ToPrimitive](https://tc39.es/ecma262/#sec-toprimitive):

> The abstract operation ToPrimitive takes argument input and optional argument PreferredType. It converts its input argument to a non-Object type. If an object is capable of converting to more than one primitive type, it may use the optional hint PreferredType to favour that type. It performs the following steps when called: